sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsRequestInfo.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public final class CosmosDiagnosticsRequestInfo {
+
+    private final String activityId;
+    private final String partitionId;
+    private final String partitionKeyRangeId;
+    private final String requestType;
+    private final Instant startTime;
+    private final Duration duration;
+    private final Duration backendLatency;
+    private final Collection<CosmosDiagnosticsRequestEvent> events;
```

Comment:
If this is always going to be ordered, should we use a list instead? Why use collection?

Reply:
Done